Core/src/org/sleuthkit/autopsy/report/ReportProgressLogger.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.report;
+
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.coreutils.Logger;
+
+/**
+ * Writes progress and status messages to the application log.
+ */
+class ReportProgressLogger implements ReportProgressPanel {
+
+    private static final Logger logger = Logger.getLogger(ReportProgressLogger.class.getName());
+    private volatile ReportProgressPanel.ReportStatus status;
+
+    ReportProgressLogger() {
+        status = ReportStatus.QUEUING;
+    }
+
+    /**
+     * Gets the current status of the generation of the report.
+     *
+     * @return The report generation status as a ReportStatus enum.
+     */
+    @Override
+    public ReportStatus getStatus() {
+        return status;
+    }
+
+    /**
+     * Logs that the report has been started.
+     */
+    @Override
+    public void start() {
+        status = ReportStatus.RUNNING;
+        logger.log(Level.INFO, "Report started");
+    }
+
+    /**
+     * NO-OP.
+     *
+     * @param max The maximum value.
+     */
+    @Override
+    public void setMaximumProgress(int max) {
+    }
+
+    /**
+     * NO-OP.
+     */
+    @Override
+    public void increment() {
+    }
+
+    /**
+     * NO-OP.
+     *
+     * @param value The value to be set.
+     */
+    @Override
+    public void setProgress(int value) {
+    }
+
+    /**
+     * NO-OP.
+     *
+     * @param indeterminate True if the progress bar should be set to
+     * indeterminate.
+     */
+    @Override
+    public void setIndeterminate(boolean indeterminate) {
+    }
+
+    /**
+     * Logs the status message.
+     *
+     * @param statusMessage String to use as label text.
+     */
+    @Override
+    public void updateStatusLabel(String statusMessage) {
+        if (status != ReportStatus.CANCELED) {
+            logger.log(Level.INFO, "Status update: {0}", statusMessage);
+        }
+    }
+
+    /**
+     * Logs the final status of the report generation.
+     *
+     * @param reportStatus The final status, must be COMPLETE or ERROR.
+     */
+    @Override
+    public void complete(ReportStatus reportStatus) {
+        if (status != ReportStatus.CANCELED) {
+            switch (reportStatus) {
+                case COMPLETE: {
+                    status = ReportStatus.COMPLETE;
+                    logger.log(Level.INFO, "Report completed");
+                    break;
+                }
+                case ERROR: {
+                    status = ReportStatus.ERROR;
+                    logger.log(Level.INFO, "Report completed with errors");
+                    break;
```

Comment:
The generic nature of the completed, cancelled, etc., errors is not ideal for log entries. I'm thinking we could make the messages less generic if we pass a report display name to the constructor, then we could have messages such as "CASE-UCO report completed", etc.

Reply:
@rcordovano That only works while we run a single report at a time. So I would cross that bridge when we get there. Perhaps for now we can just say "Reporting has completed" or "Report generation completed"?

Reply:
@eugene7646, okay. It will become moot if we get to do the resolution of the technical debt we are adding for this story, anyway.